src/Gaffer/ContextVariablesTweaks.cpp

```diff
@@ -0,0 +1,122 @@
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "Gaffer/ContextVariablesTweaks.h"
+
+#include "IECore/DataAlgo.h"
+
+using namespace IECore;
+using namespace Gaffer;
+
+GAFFER_NODE_DEFINE_TYPE( ContextVariablesTweaks );
+
+size_t ContextVariablesTweaks::g_firstPlugIndex = 0;
+
+ContextVariablesTweaks::ContextVariablesTweaks( const std::string &name ) : ContextProcessor( name )
+{
+	storeIndexOfNextChild( g_firstPlugIndex );
+	addChild( new BoolPlug( "ignoreMissing", Plug::In, false ) );
+	addChild( new TweaksPlug ( "tweaks" ) );
+}
+
+ContextVariablesTweaks::~ContextVariablesTweaks()
+{
+
+}
+
+Gaffer::BoolPlug *ContextVariablesTweaks::ignoreMissingPlug()
+{
+	return getChild<BoolPlug>( g_firstPlugIndex );
+}
+
+const Gaffer::BoolPlug *ContextVariablesTweaks::ignoreMissingPlug() const
+{
+	return getChild<BoolPlug>( g_firstPlugIndex );
+}
+
+Gaffer::TweaksPlug *ContextVariablesTweaks::tweaksPlug()
+{
+	return getChild<TweaksPlug>( g_firstPlugIndex + 1 );
+}
+
+const Gaffer::TweaksPlug *ContextVariablesTweaks::tweaksPlug() const
+{
+	return getChild<TweaksPlug>( g_firstPlugIndex + 1 );
+}
+
+bool ContextVariablesTweaks::affectsContext( const Plug *input ) const
+{
+	return tweaksPlug()->isAncestorOf( input ) || input == ignoreMissingPlug();
+}
+
+void ContextVariablesTweaks::processContext( Context::EditableScope &context, IECore::ConstRefCountedPtr &storage ) const
+{
+	const TweaksPlug *tweaksPlug = this->tweaksPlug();
+	if( tweaksPlug->children().empty() )
+	{
+		return;
+	}
+
+	const bool ignoreMissing = ignoreMissingPlug()->getValue();
+
+	IECore::CompoundDataPtr sourceData = new CompoundData();
+	IECore::CompoundDataPtr result = new CompoundData();
+
+	tweaksPlug->applyTweaks(
+		[&context, &sourceData]( const std::string &valueName )
+		{
+			DataPtr value = context.context()->getAsData( valueName, nullptr );
+			sourceData->writable()[valueName] = value;
+			return value.get();
+		},
+		[&context, &result]( const std::string &valueName, DataPtr newData )
+		{
+			if( newData == nullptr )
+			{
+				context.remove( valueName );
+				return true;
+			}
+
+			context.set( valueName, newData.get() );
+			result->writable()[valueName] = newData;
```

Comment:
It doesn't particularly matter what `valueName` is here - we just want to keep `newData` alive. We could probably do that more cheaply with `ObjectVector storageVector` rather than `CompoundData result`, which I think would also make the purpose of the variable a bit clearer.

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/5047/commits/19418b4fa6215cc4afabe83fbd08f1c8acb24563